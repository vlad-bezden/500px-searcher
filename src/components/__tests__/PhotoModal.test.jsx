import React from 'react'
import { shallow } from 'enzyme'

import PhotoModal from '../PhotoModal'
import { MOCK_500PX_PHOTOS_RESPONSE } from '../../tests/mocks'

const photo = JSON.parse(MOCK_500PX_PHOTOS_RESPONSE).photos[0]

test('PhotoModal renders correctly when there is no photo', () => {
  const component = shallow(
    <PhotoModal show={true} onHide={f => f} />
  )

  const tree = component.getNode()

  expect(tree).toMatchSnapshot()
})

test('PhotoModal renders correctly when there is a photo', () => {
  const component = shallow(
    <PhotoModal photo={photo} show={true} onHide={f => f} />
  )

  const tree = component.getNode()

  expect(tree).toMatchSnapshot()
})

test('PhotoModal calls onHide when hide button is clicked', () => {
  const onHideStub = jest.fn()
  const component = shallow(
    <PhotoModal photo={photo} show={true} onHide={onHideStub} />
  )

  component.simulate('hide')

  expect(onHideStub).toBeCalled()
})