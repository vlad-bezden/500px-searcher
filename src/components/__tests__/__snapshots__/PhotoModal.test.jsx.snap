// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoModal renders correctly when there is a photo 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  bsSize="large"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "modals": Array [],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    aria-label="Close"
    bsClass="modal-header"
    closeButton={true}
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Junction trails
       by 
      YoHomma
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <img
      src="https://drscdn.500px.org/photo/63835479/m%3D900_k%3D1_a%3D1/399d0e54f01e5d4398566b498cc1519e"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </ModalBody>
</Modal>
`;

exports[`PhotoModal renders correctly when there is no photo 1`] = `null`;
